### DISCUSSION TITLE
Routine per spegnere luci post blackout

### DISCUSSION THREAD

**Discussion by Luponr3** [indomus/forum#2397] [HUB: Home Assistant]
Ciao a tutti,
spero di non aver sbagliato sezione.
sto cercando una soluzione per fare sì che alexa spenga le luci dopo un black out.
quello che stavo pensando di fare era di impostare un controllo ogni 30 minuti se imposto (e fintanto che permane) uno stato "non in casa" che già mi triggera diverse azioni, ma non ho trovato come fare?
avete suggerimenti? in caso anche andando ad installare HA su un Raspberry.

Grazie mille

**Comment by Scialla** [comment]
Devi spiegare meglio; 
- di che luci si tratta e che tipologia di connessione hanno (wifi, zigbee, BT)? Spesso questa è una impostazione selezionabile nella configurazione delle luci stesse
- perché delegheresti questa azione ad Alexa? Hai postato nella categoria HomeAssistant quindi 'presumo' che tu lo utilizzi
- usi un UPS sulla macchina su cui utilizzi HA?  Se non lo usi puoi fare con facilità con una automazione che triggera alla partenza di HA e spegne le luci che hanno questo problema

**Reply by Luponr3** [reply]
Ciao, allora:
1. sono miste, philips hue e altre marche cinesi, ho guardato nelle impostazioni e non ho trovato una funzione che ripristini lo stato nel momento in cui torna la luce.
2. al momento non ho attivo HA. dovrei installarlo e configurare tutto. fino ad ora non ho mai esigenze così spinte da portarmi ad usarlo  (a parte averci giocato un pochettino un annetto fa)
3. no, questo potrebbe essere interessante...
grazie mille

**Reply by Scialla** [reply]
Con solo Alexa sei limitato alla sola possibilità data dalle impostazioni delle lampadine. Non penso ci sia modo di fare quello che chiedi senza utilizzare un hub serio che consente di fare automazioni limitate solo dalla fantasia dell'utente. Le routine di Alexa sono limitatissime

**Reply by Luponr3** [reply]
eh stavo vedendo, pensavo di riuscire a fare dei processi ricorsivi all'interno della singola routine,  non c'è modo (cosa aspettino a mettere queste cose è un mistero).
mi sa che devo installare nuovamente HA e vedere un po'.
ho delle guide, ma si accettano suggerimenti su come impostare poi una funzionalità come quella che suggerivi.
Grazie mille ancora intanto.

**Reply by Scialla** [reply]
Intanto devi cominciare a integrare in HA tutti i tuoi device e non è detto che sia così immediato. Se sono device zigbee che hai integrato su alexa, non riuscirai a integrarli su HA senza prendere un coordinator universale.
Per quanto riguarda l'automazione per spegnere luci in avvio è banale:
```
  - alias: Reset luci in avvio HA
    trigger:
    - platform: homeassistant
      event: start
    action:
    - service: light.turn_off
      entity_id: light.luce1, light.luce2, light.luce3
```
eventualmente aggiungendo un `delay` prima dello spegnimento nella `action` nel caso l'automazione fosse eseguita prima che HA abbia caricato i device. 

Consiglio: passa ad HA, con solo Alexa ci fai veramente poco, e lascia gestire tutte le automazioni ad HA eliminando le routine. Puoi esporre poi le entità di HA ad Alexa per gestirle vocalmente.

Leggiti un pò di guide su inDomus per farti una idea di cosa ti aspetta

**Comment by indomus** [comment]
> Ciao a tutti, spero di non aver sbagliato sezione. sto cercando una soluzione per fare sì che alexa spenga le luci dopo un black out. quello che stavo pensando di fare era di impostare un controllo ogni 30 minuti se imposto (e fintanto che permane) uno stato "non in casa" che già mi triggera diverse azioni, ma non ho trovato come fare? avete suggerimenti? in caso anche andando ad installare HA su un Raspberry.
> 
> Grazie mille

la strada maestra è quella di impostare il comportamento alla rialimentazione delle luci SULLE luci stesse, se possibile. per esempio, se integrassi le HUE (non so se si possa far anche via app HUE, ma sempre sulle luci) con [ZigBee2MQTT](https://indomus.it/componenti/zigbee2mqtt-bridge-gateway-software-zigbee/) o [ZHA](https://indomus.it/componenti/zha-zigbee-home-automation-bridge-gateway-software-zigbee/) per usarle con Home Assistant, c'è un'opzione per impostare il cosiddetto power on state, ovvero che deve fare (da sé) quando viene rialimentata.

**Reply by Luponr3** [reply]
grazie mille.
devo trovare il tempo di rimettere su l'ambiente di controllo e vedere cosa riesco a fare.
